components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/APIClient.java

```diff
@@ -221,20 +234,48 @@ private ActionInvocationErrorResponse handleErrorResponse(HttpEntity responseEnt
         return null;
     }
 
-    private <T> T deserialize(HttpEntity responseEntity, Class<T> returnType) throws ActionInvocationException {
+    private String validateJsonResponse(HttpEntity responseEntity) throws ActionInvocationException {
 
         if (!isAcceptablePayload(responseEntity)) {
             throw new ActionInvocationException("The response content type is not application/json.");
         }
 
-        ObjectMapper objectMapper = new ObjectMapper();
         try {
-            String jsonResponse = EntityUtils.toString(responseEntity);
+            return EntityUtils.toString(responseEntity);
+        } catch (IOException e) {
+            throw new ActionInvocationException("Reading JSON response failed.", e);
+        }
+    }
+
+    private ActionInvocationResponse.APIResponse deserialize(HttpEntity responseEntity)
```

Comment:
Better rename as "deserialzeSuccessOrFailureResponse" as that's what this function does

Reply:
This is updated with e01617689c045e164749dedaaf2f7f0603920ae2